### PR TITLE
CI sanity check test for env vars

### PR DESCRIPTION
**Pull request by clee2000**
Make a test that fails on purpose to trigger retries.  Check the opposite of success (that env vars exist)

It's bit hacky because I want it to fail on the normal flow in order to trigger reruns but I don't want to expose the failures to users since it's confusing.

### CODE REVIEW

File: test/run_test.py [hunk 1]
```diff
@@ -994,6 +996,23 @@ def get_pytest_args(options, is_cpp_test=False, is_distributed_test=False):
...
+    assert (
+        test.name == "test_ci_sanity_check_fail"
+    ), f"This handler only works for test_ci_sanity_check_fail, got {test.name}"
+    ret_code = run_test(test, test_directory, options, print_log=False)
+    # This test should fail
+    if ret_code != 1:
+        return 1
+    test_reports_dir = str(REPO_ROOT / "test/test-reports")
+    # Delete the log files and xmls generated by the test
+    for file in glob.glob(f"{test_reports_dir}/test_ci_sanity_check_fail*.log"):
```
Comment (huydhn) [inline test/run_test.py:1009]: Nit: I think you could use `test.name` here in the f-string

**Review by huydhn — APPROVED** [review]
LGTM!

**Comment by eqy** [comment]
@clee2000 We're seeing that this breaks on our CI that is run outside of GHA, is there an example of what the restart mechanism should look like, or how we should disable this test if it is not run on GHA?

**Comment by clee2000** [comment]
> @clee2000 We're seeing that this breaks on our CI that is run outside of GHA, is there an example of what the restart mechanism should look like, or how we should disable this test if it is not run on GHA?

The changes in this PR in run_test.py show what the restart mechanism should look like.  I would recommend not running this file at all if not run via run_test.py

**Pull request closed without merging**